src/iop/demosaic.c

```diff
@@ -2480,6 +2485,48 @@ static void passthrough_monochrome(float *out, const float *const in, dt_iop_roi
   }
 }
 
+static void blk4_monochrome(float *out, const float *const in, dt_iop_roi_t *const roi_out,
+                                   const dt_iop_roi_t *const roi_in, const uint32_t filters, const uint32_t rgb_filter)
+{
+  assert(roi_in->width >= roi_out->width);
+  assert(roi_in->height >= roi_out->height);
+
+  const float rgb_pars[4][4] = {
+    { 0.2225f, 0.7169f / 2.0f, 0.0606f, 0.0f }, // luminance
```

Comment:
that luminance formula is valid only for a particular RGB space. Camera RGB can be absolutely whatever. A real luminance should read the coefficients of colorin ahead, and apply them there.

Reply:
right. I changed those to more abstract numbers.